opentelemetry-sdk/src/trace/evicted_hash_map.rs

```diff
@@ -137,7 +137,7 @@ impl<'a> IntoIterator for &'a EvictedHashMap {
 }
 
 /// An iterator over the entries of an `EvictedHashMap`.
-#[derive(Debug)]
+#[derive(Debug, Clone)]
```

Comment:
What's the motivation here?

Reply:
To be able to operate over a collection of attributes generically, cloning the iterator multiple times so the clones can be mutated.

```rust
pub(crate) fn get_attributes_as_json<'a, C>(attribs: C) -> String
where C: Iterator<Item = (&'a Key, &'a Value)>
{
    do_thing_with_attributes(attribs.clone());
    do_a_second_thing_with_attributes(attribs));
}
```

The underlying iterator type is Clone, and the OrderedMap iterator for Log attributes is Clone. Having the wrapper struct implement Clone shouldn't be impactful except through its absence.

Reply:
Okay, I guess. Seems pretty niche?